web/assets/init.js

```diff
@@ -0,0 +1,59 @@
+if ("serviceWorker" in navigator) {
+    window.addEventListener("load", () => {
+        navigator.serviceWorker
+            .register("/service-worker.js", { scope: "/" })
+            .catch((err) => console.error("Service Worker Failed to Register", err));
+    });
+}
+
+/* put this in document head to avoid FOUC */
+const mediaQueryPrefersDarkScheme = window.matchMedia("(prefers-color-scheme: dark)");
+function updateTheme() {
+    const shouldBeDark =
+        localStorage.themeMode === "dark" ||
+        (localStorage.themeMode === "system" && mediaQueryPrefersDarkScheme.matches);
+    if (document.documentElement.classList.contains("dark") !== shouldBeDark) {
+        document.documentElement.classList.toggle("dark");
+    }
+}
+const setThemeMode = (mode) => {
+    localStorage.themeMode = mode;
+};
+const getThemeMode = () => localStorage.themeMode;
+
+// transition from old dark theme system
+if ("darkTheme" in localStorage) {
+    /* TODO: send notification? */
```

Comment:
I am not a fan of `TODO` comments.
Imo the whole if clause could be dropped completely without a huge loss to the user